library/src/main/java/com/mapzen/android/MapzenRouter.java

```diff
@@ -20,51 +20,39 @@
   private static final String TAG = MapzenRouter.class.getSimpleName();
 
   private Router internalRouter = new ValhallaRouter();
-  private Context context;
 
-  private static final HashMap<DistanceUnits, Router.DistanceUnits> UNITS_TO_ROUTER_UNITS =
-      new HashMap<>();
-  static {
-    UNITS_TO_ROUTER_UNITS.put(DistanceUnits.MILES, Router.DistanceUnits.MILES);
-    UNITS_TO_ROUTER_UNITS.put(DistanceUnits.KILOMETERS, Router.DistanceUnits.KILOMETERS);
+  /**
+   * Creates a new {@link MapzenRouter} with api key set from mapzen.xml.
+   */
+  public MapzenRouter(Context context) {
+    String apiKey = initializeRouterKey(context);
+    TurnByTurnHttpHandler httpHandler = new TurnByTurnHttpHandler();
+    httpHandler.setApiKey(apiKey);
+    internalRouter.setHttpHandler(httpHandler);
   }
 
   /**
-   * Creates a new {@link MapzenRouter}.
-   * @param context
+   * Creates a new {@link MapzenRouter} with api key set in code.
    */
-  public MapzenRouter(Context context) {
-    this.context = context;
-    initializeRouterKey();
-    initializeRouterUrl();
+  public MapzenRouter(String apiKey) {
+    TurnByTurnHttpHandler httpHandler = new TurnByTurnHttpHandler();
+    httpHandler.setApiKey(apiKey);
+    internalRouter.setHttpHandler(httpHandler);
   }
 
-  private void initializeRouterKey() {
+  private String initializeRouterKey(Context context) {
```

Comment:
What do you think about moving the routing API key initialization into `AndroidModule` so the `TurnByTurnHttpHandler` can be injected with Dagger? This is how we are currently creating and using the `TileHttpHandler`?

https://github.com/mapzen/android/blob/master/library/src/main/java/com/mapzen/android/dagger/AndroidModule.java#L51

This way all the API key initialization would be in the same place. This would also reduce the need for reflection in the tests.